packages/lit-helpers/src/read-only-properties.d.ts

```diff
@@ -0,0 +1,27 @@
+// when we update to typescript 3.8
+// import type { LitElement } from "lit-element";
+import { LitElement } from "lit-element";
+
+declare type Constructor<T = {}> = new (...args: any[]) => T;
+
+declare module 'lit-element/lib/updating-element' {
+  interface PropertyDeclaration {
+    readOnly?: boolean;
+  }
+}
+
+export interface ReadOnlyClass {
+  /**
+   * Set read-only properties
+   */
+  setReadOnlyProperties(props: {
```

Comment:
Shoud this be a public method?

Reply:
Should probably be protected, but ts doesn't support that yet, so the best we could do would be a jsdoc `@protected` tag

Reply:
Do we want to follow the convention of protected with a leading `_` ?

Reply:
None of LitElement's protected methods are prefixed so I'd prefer not to here

Reply:
hmmm, 50/50 on this one so I'll defer to your preference :)